packages/grafana-ui/src/components/Layout/Grid/Grid.internal.story.tsx

```diff
@@ -0,0 +1,360 @@
+import { Meta, StoryFn } from '@storybook/react';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+
+import { useTheme2 } from '../../../themes';
+import { Stack } from '../../../unstable';
+import { withCenteredStory } from '../../../utils/storybook/withCenteredStory';
+
+import { Grid } from './Grid';
+import mdx from './Grid.mdx';
+import { GridItem } from './GridItem';
+
+const gridItem = (index: number, theme: GrafanaTheme2, width?: string, height?: string) => {
+  const colorOrder = index >= 6 ? index - 6 : index;
+  const itemOrder = index + 1;
+  const colorList = ['red', 'orange', 'purple', 'blue', 'yellow', 'green'].map(theme.visualization.getColorByName);
+  return (
+    <div
+      key={index}
+      style={{
+        backgroundColor: colorList[colorOrder],
+        width,
+        height,
+        fontSize: theme.typography.bodySmall.fontSize,
+        color: theme.colors.text.maxContrast,
+      }}
+    >
+      Item {itemOrder}
+    </div>
+  );
+};
+
+const meta: Meta<typeof Grid> = {
+  title: 'General/Layout/Grid',
+  component: Grid,
+  decorators: [withCenteredStory],
+  parameters: {
+    docs: {
+      page: mdx,
+    },
+  },
+  argTypes: {
+    templateColumns: {
+      control: {
+        type: 'text',
+      },
+    },
+    templateRows: {
+      control: {
+        type: 'text',
+      },
+    },
+    autoFlow: {
+      control: {
+        type: 'select',
+        options: ['row', 'column', 'row dense', 'column dense'],
+      },
+    },
+  },
+  args: {
+    display: 'grid',
+    gap: 1,
+  },
+};
+
+export const Basic: StoryFn<typeof Grid> = (args) => {
```

Comment:
Some options of the autoFlow prop look the same (row = row dense, column = column dense). Can you please check whether it's possible to make clear what the differences are?

Reply:
It seems it reorders the items to try to fill the holes first so the smaller items go last. It is explained [here](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-grid-auto-flow). Will it be useful to show this as an example? This is a normal property of CSSGrid 🤷